esphomeRemote/esphomeRemoteLight.h

```diff
@@ -2,30 +2,172 @@
+      subscribe_homeassistant_state(&LightService::max_mireds_changed, newEntityId.c_str(),"max_mireds");
+      subscribe_homeassistant_state(&LightService::brightness_changed, newEntityId.c_str(),"brightness");
+      subscribe_homeassistant_state(&LightService::color_temp_changed, newEntityId.c_str(),"color_temp");
+      subscribe_homeassistant_state(&LightService::color_mode_changed, newEntityId.c_str(),"color_mode");
     }
     std::string friendlyName;
     std::string entityId;
     DisplayUpdateInterface& display;
+    int brightness = 0;
+    int color_temp = 0;
+    std::string color_mode = "";
     bool onState;
+    int min_mireds = 0;
+    int max_mireds = 0;
+
+    void decTemperature() {
+        std::stringstream ss;
+        ss << (color_temp - id(inc_color_temperature_step));
+        const std::map< std::string, std::string > data = {
+                                                              {"entity_id",entityId.c_str()},
+                                                              {"color_temp", ss.str()} ,
+                                                          };
+        setAttribute(data);
+    }
+
+    void incTemperature() {
+        std::stringstream ss;
+        ss << (color_temp + id(dec_color_temperature_step));
+        const std::map< std::string, std::string > data = {
+                                                              {"entity_id",entityId.c_str()},
+                                                              {"color_temp", ss.str()} ,
+                                                          };
+        setAttribute(data);
+    }
+
+    void decBrightness() {
+        std::stringstream ss;
+        ss << (brightness - id(dec_brightness_step));
+        const std::map< std::string, std::string > data = {
+                                                              {"entity_id",entityId.c_str()},
+                                                              {"brightness", ss.str()} ,
+                                                          };
+        setAttribute(data);
+    }
+
+    void incBrightness() {
+        std::stringstream ss;
+        ss << (brightness + id(inc_brightness_step));
+        const std::map< std::string, std::string > data = {
+                                                              {"entity_id",entityId.c_str()},
+                                                              {"brightness", ss.str()} ,
+                                                          };
+        setAttribute(data);
+    }
+
+    void setAttribute(const std::map< std::string, std::string> &data){
+        call_homeassistant_service("light.turn_on", data);
+    }
 
     void toggleLight() {
       call_homeassistant_service("light.toggle", {
         {"entity_id", entityId.c_str()}
       });
     }
 
+    bool supportsBrightness(){
+        return color_mode.compare("onoff") != 0 && color_mode.compare("unknown") != 0;

```

Comment:
in other classes i was using strcmp to compare strings `if(strcmp(state.c_str(), "playing")`